src/test/scala/adventures/observable/ObservableAdventuresSpec.scala

```diff
@@ -74,19 +74,22 @@ class ObservableAdventuresSpec extends Specification {
     }
 
     "handle a paginated feed" in {
-      val pages = Map(
-        PageId.FirstPage -> PaginatedResult(List(SourceRecord("1", "1.1")), Some(PageId("2"))),
-        PageId("2") -> PaginatedResult(List(SourceRecord("2", "2.2")), Some(PageId("3"))),
-        PageId("3") -> PaginatedResult(List(SourceRecord("3", "3.3")), None)
-      )
+      val lastPage = 1000
+      val pages = (0 to lastPage).map { n =>
+        val pageRecords = List(SourceRecord(n.toString, s"$n.$n"))
+        val nextPage = if (n >= lastPage) None else Some(PageId((n + 1).toString))
+
+        PageId(n.toString) -> PaginatedResult(pageRecords, nextPage)
+      }.toMap
 
       def readPage(pageId: PageId): Task[PaginatedResult] = {
         Task(pages(pageId))
       }
 
       val obs = ObservableAdventures.readFromPaginatedDatasource(readPage)
 
-      val expected = List(SourceRecord("1", "1.1"), SourceRecord("2", "2.2"), SourceRecord("3", "3.3"))
+      val expected = pages.values.toList.flatMap(_.results)
+
       runLog(obs) must beEqualTo(expected)
```

Comment:
Now that there are 1000 pages, if the test fails is the console output huge?

Wondering if we should keep the existing test as is with smaller output and add a new test "not blow the stack when handling a large paginated feed"